core/container.go

```diff
@@ -74,6 +74,10 @@ type Container struct {
 	// Services to start before running the container.
 	Services    ServiceBindings `json:"services,omitempty"`
 	HostAliases []HostAlias     `json:"host_aliases,omitempty"`
+
+	// Focused indicates whether subsequent operations will be
+	// focused, i.e. shown more prominently in the UI.
+	Focused bool `json:"focused"`
```

Comment:
Not a blocker but it doesn't feel ideal that this would change the ID. In the context of Zenith this matters more since once we support `Container`/`Directory`/etc. as input args to entrypoints, the execution will be invalidated just by changing `Focused`, which is more of a UX concern.

However, there's a few other things like this already (e.g. pipeline labels), so this doesn't really make it any worse. We'll need to come up with some generic way of dealing with all these at some point in the future.

If there's some utterly trivial different approach to this that doesn't involve putting it in the ID, then I'd say go for it right away, but I'm guessing there isn't anything obvious yet, in which case all good for now.

Reply:
Yeah I'd also like to find a generic alternative to this pattern. Using queries for "meta" operations like this also blew up in my face when I was trying to come up with a proposal for generic query memoization (as an approach to dependency pinning):

https://github.com/dagger/dagger/pull/5420/files#diff-7340b8a6d2c5b52c935f75ec72038d28b8c606323ea41bd29ced4b521eae7761R115-R118

(Quoting in case that link breaks with future pushes:)

> * **TODO:** We'll need some way to prevent meta-queries like `pipeline()` and
  `memoize()` from becoming part of the cache key. If we can't come up with a
  better idea, we could always just reserve the words and drop them from the
  query path during memoization.

I think this is OK for now since it's a drop in the bucket compared to storing pipeline info (which also massively inflate the ID values, as I'm sure you're already aware :grin:).